src/controllers/colormapper.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <QColor>
+#include <QMap>
+#include <QObject>
+#include <QPair>
+#include <QVariant>
+
+#include "util/assert.h"
+
+class ColorMapper final {
+  public:
+    ColorMapper() = delete;
+    ColorMapper(const QMap<QRgb, QVariant> availableColors)
+            : m_availableColors(availableColors) {
+        DEBUG_ASSERT(!m_availableColors.isEmpty());
```

Comment:
because we assert if availablecolors is empty, we don't need to check those cases again in getnearestcolor.  The class will never be created with an empty map.

Reply:
We don't check it there except maybe in the `nearestColorDistance < qInf()` assertion, and that also checks the color matching code, too.